ui/pages/confirm-transaction/confirm-transaction.component.js

```diff
@@ -49,27 +52,34 @@ export default class ConfirmTransaction extends Component {
     setDefaultHomeActiveTabName: PropTypes.func,
   };
 
+  constructor(props) {
+    super(props);
+    this.state = {};
+  }
+
   componentDidMount() {
     const {
       totalUnapprovedCount = 0,
       sendTo,
       history,
       mostRecentOverviewPage,
       transaction: { txParams: { data, to } = {} } = {},
-      fetchBasicGasEstimates,
       getContractMethodData,
       transactionId,
       paramsTransactionId,
       getTokenParams,
       isTokenMethodAction,
     } = this.props;
 
+    getGasFeeEstimatesAndStartPolling().then((pollingToken) => {
```

Comment:
The same question I raised elsewhere about cancelling polling can be raised here.

Reply:
I added a instance variable _isMounted where this occurs, and if _isMounted if alse after the promise resolves we disconnect instead of setting the pollingToken in state.